File: kq-projection-dotnet/kq-projection/Common/Models/Events/ExceptionalEventData.cs
using KQ_Projection.Common.Enums;
using System.Text.Json.Serialization;

namespace KQ_Projection.Common.Models.Events;
public class ExceptionalEventData : EventData
{
    [JsonPropertyName("previousEventType")]
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public EventType previousEventType { get; set; }

    [JsonPropertyName("exceptionMessage")]
    public String exceptionMessage { get; set; }
}

File: java-apps/kq-common/src/main/java/com/kafkaquest/kq/common/util/models/events/eventdata/ExceptionalEventData.java
package com.kafkaquest.kq.common.util.models.events.eventdata;

import com.kafkaquest.kq.common.util.models.events.enums.EventType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@EqualsAndHashCode(callSuper = false)
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ExceptionalEventData extends EventData {

    private EventType previousEventType;

    private String exceptionMessage;

}
